src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -60,9 +60,20 @@ class AppsResource @Inject() (
              @DefaultValue("false")@QueryParam("force") force: Boolean): Response = {
 
     val now = clock.now()
+
+    val bodyJson = Json.parse(body)
+    if (bodyJson.toString() == "[]")
+      throw new BadRequestException("Please specify an AppDefinition to create")
+
     val app =
-      validateApp(Json.parse(body).as[V2AppDefinition].withCanonizedIds().toAppDefinition)
-        .copy(versionInfo = AppDefinition.VersionInfo.OnlyVersion(now))
+      bodyJson.asOpt[V2AppDefinition] match {
```

Comment:
If I am not mistaken, we got `JsResultExceptions` before this change which had special treatment in `MarathonExceptionMapper` and actually indicating detail on the errors.

What is it that you would like to improve specifically?